src/harrypotterplugin/spells/StuporSpell.java

```diff
@@ -7,12 +7,14 @@
 import harrypotterplugin.utilities.Spell;
 import harrypotterplugin.utilities.SpellProjectile;
 
+import java.util.Objects;
+
 public class StuporSpell extends Spell {
 
 	public StuporSpell() {
 		super("Stupor", SpellType.CHARM, 2);
 		setSpellCastAction(event -> {
-			SpellProjectile spellProjectile = new SpellProjectile(1, 40, event.getPlayer().getEyeLocation(), event.getPlayer().getEyeLocation().getDirection());
+			SpellProjectile spellProjectile = new SpellProjectile(1, 40, Objects.requireNonNull(event.getPlayer()).getEyeLocation(), event.getPlayer().getEyeLocation().getDirection());
```

Comment:
Again the same issue this crashes with an NPE, but the player can never be null in this case